elastic/kibana
# [Security Solution] Remove validation requiring action id to be UUID

## Summary

Closes https://github.com/elastic/kibana/issues/105271

Validation in the security solution was requiring the connector ID to be a UUID, but this is not actually a requirement for all valid connectors.

### Testing
1. Create index `test-action` for your test connector to write to
1. Add a preconfigured connector to your kibana.dev.yml
```
xpack:
  actions:
    preconfigured:
      preconfiguredindexaction:
        actionTypeId: .index
        config:
          index: 'test-action'
          refresh: false
        name: preconfigured-index-action
```
3. Create a rule that uses this connector, observe that the "No connector selected" error no longer appears
4. Trigger the rule's condition so an alert gets written and the connector is used as expected
